lib/reporter.js

```diff
@@ -110,6 +110,17 @@ class AllureReporter extends events.EventEmitter {
             this.currentHooks.pop();
         })
 
+        this.on('runner:screenshot', (screenshot) => {
+
+            var filename = this.getScreenshotFilename(screenshot),
+                data = new Buffer(screenshot.data, 'base64')
+
+            this.print('screenshot', filename)
+            this.allure.addAttachment(filename, data)
```

Comment:
We are using [file-type](https://github.com/sindresorhus/file-type) for auto-detect file extention if `type` argument is empty.

>Do you have a suggestion for better auto-naming? 
The best option will be `filename` value of the argument from [saveScreenshot](http://webdriver.io/api/utility/saveScreenshot.html) command